cypress/e2e/immutableDatabaseTests/linkCheck.ts

```diff
@@ -0,0 +1,104 @@
+import { setTokenUserViewPort } from '../../support/commands';
+
+function formatPaths(paths: string[]) {
+  let formattedPaths = '';
+  paths.forEach((path) => {
+    formattedPaths += '\n- ' + path;
+  });
+  return formattedPaths;
+}
+
+describe('Find broken anchor links', () => {
+  setTokenUserViewPort();
+
+  function checkUrls(path: string, selector: string) {
+    let brokenUrls = [];
+    cy.visit(path);
+    cy.get(selector)
+      .find('a')
```

Comment:
Similar question to Steve, why are we only checking certain selectors?

If we check all links on a page, this could be simplified to:

```
cy.visit(path);
cy.get('a').each(...your  URL checking code...);
```

Reply:
One reason was to prevent checking navbar/footer links multiple times.

Another reason was to filter out "dynamic" content/links. For example, at /dashboard, the test could avoid checking the links to /my-workflows/..., /my-tools/..., etc in the entry boxes, or the links in Mastodon posts, which might be less valuable to us?

Reply:
> One reason was to prevent checking navbar/footer links multiple times.

This was the intent of my comment about keeping track of already visited images and links (I'd only said images, which was confusing).

1. You get rid of the selector
2. You navigate to `path`
3.  You visit all links/images in the document, keeping track in a set of the ones you have visited.
4. You go to a new path
5. You get all links in the document, remove the ones you have visited, then check the status of the remaining ones.

That way you don't need the selector, need to know which component corresponds to each path, etc.

On the dynamic content links, not sure, maybe you could exclude them?